source/Octopus.Tentacle.Tests.Integration/Support/TentacleExeFinder.cs

```diff
@@ -1,40 +1,66 @@
 using System;
 using System.IO;
-using NUnit.Framework;
-using NUnit.Framework.Internal;
 using Octopus.Tentacle.Util;
 
 namespace Octopus.Tentacle.Tests.Integration.Support
 {
     public class TentacleExeFinder
     {
-
         public static string FindTentacleExe()
+        {
+            return FindTentacleExe(TentacleRuntime.Default);
+        }
+
+        public static string FindTentacleExe(TentacleRuntime version)
         {
             var assemblyDir = new DirectoryInfo(Path.GetDirectoryName(typeof(TentacleExeFinder).Assembly.Location)!);
             var tentacleExe = Path.Combine(assemblyDir.FullName, "Tentacle");
             
 
             // We don't have access to any teamcity environment variables so instead rely on the path. 
-            if (IsRunningInTeamCity())
+            if (TeamCityDetection.IsRunningInTeamCity())
             {
                 // Example current directory of assembly.
                 // /opt/TeamCity/BuildAgent/work/639265b01610d682/build/outputs/integrationtests/net6.0/linux-x64
                 // Desired path to tentacle.
                 // /opt/TeamCity/BuildAgent/work/639265b01610d682/build/outputs/tentaclereal/tentacle/Tentacle
+                
+            string tentacleDir =
+                version switch
+                {
+                    TentacleRuntime.Framework48 => "tentaclereal-net48",
+                    TentacleRuntime.DotNet6 => "tentaclereal-net6.0",
+                    TentacleRuntime.Default => 
+                #if NETFRAMEWORK
```

Comment:
nit: may be github review page but nesting of the .Defaut switch is a bit difficult to read